iceberg/iceberg-handler/src/test/queries/positive/iceberg_major_compaction_partitioned.q

```diff
@@ -0,0 +1,38 @@
+-- SORT_QUERY_RESULTS
+-- Mask neededVirtualColumns due to non-strict order
+--! qt:replace:/(\s+neededVirtualColumns:\s)(.*)/$1#Masked#/
+
+set hive.llap.io.enabled=true;
+set hive.vectorized.execution.enabled=true;
+set hive.optimize.shared.work.merge.ts.schema=true;
+
+create table ice_orc (
+    first_name string, 
+    last_name string
+ )
+partitioned by (dept_id bigint)
+stored by iceberg stored as orc 
+tblproperties ('format-version'='2');
+
+insert into ice_orc VALUES ('fn1','ln1', 1);
+insert into ice_orc VALUES ('fn2','ln2', 1);
+insert into ice_orc VALUES ('fn3','ln3', 1);
+insert into ice_orc VALUES ('fn4','ln4', 1);
+insert into ice_orc VALUES ('fn5','ln5', 2);
+insert into ice_orc VALUES ('fn6','ln6', 2);
+insert into ice_orc VALUES ('fn7','ln7', 2);
+
+update ice_orc set last_name = 'ln1a' where first_name='fn1';
+update ice_orc set last_name = 'ln2a' where first_name='fn2';
+update ice_orc set last_name = 'ln3a' where first_name='fn3';
+update ice_orc set last_name = 'ln4a' where first_name='fn4';
+update ice_orc set last_name = 'ln5a' where first_name='fn5';
+update ice_orc set last_name = 'ln6a' where first_name='fn6';
+update ice_orc set last_name = 'ln7a' where first_name='fn7';
+
+delete from ice_orc where last_name in ('ln1a', 'ln2a', 'ln7a');
+
+explain alter table ice_orc COMPACT 'major' and wait;
+alter table ice_orc COMPACT 'major' and wait;
+
+select * from ice_orc;
```

Comment:
what do we test here? should we print the `snapshot-summary` to check there are no delete files?

Reply:
Added printing of `snapshot-summary` to the tests. Also updated `HiveIcebergStorageHandler` to allow IOW after partition evolution in case of Iceberg Compaction.